be/src/column/column_helper.cpp

```diff
@@ -314,4 +315,40 @@ ColumnPtr ColumnHelper::convert_time_column_from_double_to_str(const ColumnPtr&
     return res;
 }
 
+bool ChunkSlice::empty() const {
+    return !chunk || offset == chunk->num_rows();
+}
+
+size_t ChunkSlice::rows() const {
+    return chunk->num_rows() - offset;
+}
+
+void ChunkSlice::reset(vectorized::ChunkUniquePtr input) {
+    chunk = std::move(input);
+}
+
+size_t ChunkSlice::skip(size_t skip_rows) {
+    size_t real_skipped = std::min(rows(), skip_rows);
+    offset += real_skipped;
+    if (empty()) {
```

Comment:
should check offset less than chunk->num_rows()  
or  in `ChunkSlice::empty()`, should check `offset >= chunk->num_rows()`;  ?

Reply:
The `ChunkSlice::empty` is implemented as `offset == chunk->num_rows()`, which could guarantee that `offset` would not exceed `num_rows()`.